### PR TITLE
Add tests for rectificationPeriod and item.address related fields

### PR DESCRIPTION
**Pull request by OrysiaDrabych**
This change is [Reviewable](https://reviewable.io/reviews/openprocurement/robot_tests/742)

### CODE REVIEW

File: op_robot_tests/tests_files/openProcedure.robot [hunk 1]
```diff
@@ -284,122 +329,246 @@ ${ITEM_MEAT}        ${True}
...
+
+Відображення дати завершення періоду редагування лоту
+  [Tags]   ${USERS.users['${viewer}'].broker}: Відображення основних даних тендера
+  ...      viewer
+  ...      ${USERS.users['${viewer}'].broker}
+  ...      tender_view  level2
+  Отримати дані із поля rectificationPeriod.endDate тендера для усіх користувачів
+
+
+Перевірити тривалість періоду редагування лоту
```
Comment (kosaniak) [inline op_robot_tests/tests_files/openProcedure.robot]: Можливість перевірити...

File: op_robot_tests/tests_files/brokers/openprocurement_client.robot [hunk 2]
```diff
@@ -238,6 +245,15 @@ Library  openprocurement_client.utils
   Set_To_Object   ${USERS.users['${username}'].tender_data}   ${fieldname}   ${fieldvalue}
 
 
+Редагувати ПДВ
+  [Arguments]  ${username}  ${tender_uaid}
+  ${tender}=  openprocurement_client.Пошук тендера по ідентифікатору  ${username}  ${tender_uaid}
+  Set_to_object  ${tender.data.value}  valueAddedTaxIncluded  True
+  Set_to_object  ${tender.data.minimalStep}  valueAddedTaxIncluded  True
```
[the remark this answers is not in the document]
Comment (OrysiaDrabych) [reply op_robot_tests/tests_files/brokers/openprocurement_client.robot]: Не буде. Створюємо процедуру з False

File: op_robot_tests/tests_files/openProcedure.robot [hunk 3]
```diff
@@ -186,6 +186,51 @@ ${ITEM_MEAT}        ${True}
   Звірити відображення поля description усіх предметів для усіх користувачів
 
 
+Відображення назви нас. пункту доставки активів лоту
+  [Tags]   ${USERS.users['${viewer}'].broker}: Відображення номенклатури тендера
+  ...      viewer
+  ...      ${USERS.users['${viewer}'].broker}
+  ...      tender_view  level2
+  ...      non-critical
```
[the remark this answers is not in the document]
Comment (OrysiaDrabych) [reply op_robot_tests/tests_files/openProcedure.robot]: :)

**Comment by mykhaly** [comment]
Review status: 0 of 9 files reviewed at latest revision, 22 unresolved discussions.

---

*[op_robot_tests/tests_files/initial_data.py, line 32 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5OdLqzuODpSghP82Lt:-L5OdLqzuODpSghP82Lu:boig5m0) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/initial_data.py#L32)):*
> ```Python
> 
> def create_fake_tenderAttempts(attempt):
>     number = [1,2,3,4]
> ```

Після ком потрібно ставити пробіл.

---

*[op_robot_tests/tests_files/initial_data.py, line 41 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5OdeP3aK1_DDuX_BWm:-L5OdeP4H7R7Qc7N7p1o:bln9vhr) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/initial_data.py#L41)):*
> ```Python
>             u'UKR': u"[ТЕСТУВАННЯ] {}".format(fake.title()),
>             u'RUS': u"[ТЕСТИРОВАНИЕ] {}".format(fake_ru.sentence(nb_words=2)),
>             u'ENG': u"[TESTING] {}".format(fake_en.sentence())
> ```

Російською речення буде складатись з двох слів, тут так не потрібно?

---

*[op_robot_tests/tests_files/initial_data.py, line 48 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5Odpi59WHOR5zDfy79:-L5Odpi59WHOR5zDfy7A:b-o2euop) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/initial_data.py#L48)):*
> ```Python
> def create_fake_description(language):
>     description = {
>             u'ua': fake.sentence(nb_words=10, variable_nb_words=True),
> ```

Можливо, варто для `create_fake_title` і `create_fake_description` очікувати на однакові значення параметру `language`?

---

*[op_robot_tests/tests_files/initial_data.py, line 60 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXgYZHTRBkFCPhS-r10-60:-L5OeaVLKweZjALrX6cH:b-216jmt) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/initial_data.py#L60)):*
<details><summary><i>Previously, ivanka12 wrote…</i></summary><blockquote>

поправ відступи
</blockquote></details>

Для перетворення днів у секунди можна використати пакет `datetime`.
Детальніше - тут: https://www.safaribooksonline.com/library/view/python-cookbook-3rd/9781449357337/ch03s12.html

---

*[op_robot_tests/tests_files/initial_data.py, line 177 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5Of-I38q3pl6DEYyOc:-L5Of-I38q3pl6DEYyOd:b-79e1x) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/initial_data.py#L177)):*
> ```Python
> 
>     data['rectificationPeriod'] = {
>             "endDate": (get_now() + timedelta(minutes=(random.randint(5,19)*1440)/accelerator)).isoformat()
> ```

Треба розставити пробіли по пеп-8.

---

*[op_robot_tests/tests_files/service_keywords.py, line 70 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5OfEIAM0skiF30-5tl:-L5OfEIAM0skiF30-5tm:b5hfamc) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/service_keywords.py#L70)):*
>     if delta < seconds:
>         return False
>     return True
можна замінити на `return delta >= seconds`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests/742)*

**Comment by OrysiaDrabych** [comment]
Review status: 0 of 9 files reviewed at latest revision, 22 unresolved discussions.

---

*[op_robot_tests/tests_files/base_keywords.robot, line 79 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggj-r10-79:-L6IkQspj3HQkQdQSSpl:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/base_keywords.robot#L79)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

Run Keyword And Expect Error
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/base_keywords.robot, line 89 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggj-r10-89:-L6IkTrXK08cSwHkBKZY:b-4si91t) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/base_keywords.robot#L89)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

Run As
</blockquote></details>

Require Failure як і Run As теж має працювати для майданчиків

---

*[op_robot_tests/tests_files/initial_data.py, line 34 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXgYZHTRBkFCPhS-r10-34:-L6Il4r6M4XTOYV7dYzn:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/initial_data.py#L34)):*
<details><summary><i>Previously, ivanka12 wrote…</i></summary><blockquote>

Для майданчиків це не підходить, для них 2,3,4 це Лоти виставляються повторно, а 1 Лоти виставляються вперше, до цього були притензії на акредитаціях. треба поміняти!
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/keywords.robot, line 758 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXgYZHTRBkFCPhW-r10-758:-L6Il9KHYS_GhHbNd-8m:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/keywords.robot#L758)):*
<details><summary><i>Previously, ivanka12 wrote…</i></summary><blockquote>

можливо варто замінити на wait_and_write_to_console?
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 194 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggk-r10-194:-L6IlAe4yt1Qqd49f9eP:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/openProcedure.robot#L194)):*
<details><summary><i>Previously, OrysiaDrabych wrote…</i></summary><blockquote>

:)
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 337 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggk-r10-337:-L6IlBnNf4yjG7Qy26vP:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/openProcedure.robot#L337)):*
<details><summary><i>Previously, ivanka12 wrote…</i></summary><blockquote>

а тут  чому 2 рівень?
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 341 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggk-r10-341:-L6IlCJwkrdf9pG9Nljq:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/openProcedure.robot#L341)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

Можливість перевірити...
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 358 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggk-r10-358:-L6IlCyyWDBS0cUR_MWz:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/openProcedure.robot#L358)):*
<details><summary><i>Previously, ivanka12 wrote…</i></summary><blockquote>

тут також чому 2 рівень?
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 379 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggk-r10-379:-L6IlD_Ksq5vH4HKn7yB:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/openProcedure.robot#L379)):*
<details><summary><i>Previously, ivanka12 wrote…</i></summary><blockquote>

тут чому 3 рівень?
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 487 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggk-r10-487:-L6IlG2jGAwctLwFFLtG:bes1zbu) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/openProcedure.robot#L487)):*
<details><summary><i>Previously, ivanka12 wrote…</i></summary><blockquote>

а видалити зі словника поле  tenderAttempts не треба?
</blockquote></details>

В тест кейсі на відображення видаляється

---

*[op_robot_tests/tests_files/openProcedure.robot, line 509 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggk-r10-509:-L6IlT-a4F5MpAA_REAs:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/openProcedure.robot#L509)):*
<details><summary><i>Previously, ivanka12 wrote…</i></summary><blockquote>

тут також відповідно
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 918 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggk-r10-918:-L6IlTteS3BXuLLOBTKh:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/openProcedure.robot#L918)):*
<details><summary><i>Previously, ivanka12 wrote…</i></summary><blockquote>

тут також тег критичний
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 930 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggk-r10-930:-L6IlULjAfVt8yArsLkh:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/openProcedure.robot#L930)):*
<details><summary><i>Previously, ivanka12 wrote…</i></summary><blockquote>

і тут
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 938 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggk-r10-938:-L6IlUeiwRmxBMBgDHgS:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/openProcedure.robot#L938)):*
<details><summary><i>Previously, ivanka12 wrote…</i></summary><blockquote>

і тут
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests/742)*

File: op_robot_tests/tests_files/brokers/openprocurement_client.robot (hunk 2 above)
[the remark this answers is not in the document]
Comment (ivanka12) [reply op_robot_tests/tests_files/brokers/openprocurement_client.robot]: де відповідь?

[the remark this answers is not in the document]
Comment (ivanka12) [reply op_robot_tests/tests_files/brokers/openprocurement_client.robot]: @OrysiaDrabych

File: op_robot_tests/tests_files/keywords.robot [hunk 4]
```diff
@@ -746,6 +746,19 @@ Require Failure
...
+  [Arguments]  ${username}
+  Дочекатись дати  ${USERS.users['${username}'].tender_data.data.rectificationPeriod.endDate}
+  Оновити LAST_MODIFICATION_DATE
+  Дочекатись синхронізації з майданчиком  ${username}
+
+
+Дочекатись дати
+  [Arguments]  ${date}
+  ${sleep}=  wait_and_write_to_console  ${date}
+  Run Keyword If  ${sleep} > 0  Sleep  ${sleep}
```
Comment (ivanka12) [inline op_robot_tests/tests_files/keywords.robot]: для чого цей сліп, якщо функція wait_and_write_to_console вже виконує очікування і нічого не повертає?

File: op_robot_tests/tests_files/openProcedure.robot [hunk 5]
```diff
@@ -719,6 +884,64 @@ ${ITEM_MEAT}        ${True}
...
+  ...      confirm_second_bid
+  [Teardown]  Оновити LAST_MODIFICATION_DATE
+  Можливість підтвердити цінову пропозицію учасником ${provider1}
+
+
+Можливість підтвердити цінову пропозицію після зміни умов третьому учаснику
+  [Tags]   ${USERS.users['${provider1}'].broker}: Подання пропозиції
+  ...      provider2
+  ...      ${USERS.users['${provider1}'].broker}
+  ...      confirm_third_bid
```
Comment (ivanka12) [inline op_robot_tests/tests_files/openProcedure.robot]: provider1 заміни на provider2

**Comment by selurvedu** [comment]
Reviewed 1 of 1 files at r1, 2 of 2 files at r2, 8 of 8 files at r20, 1 of 1 files at r21, 4 of 4 files at r22, 5 of 5 files at r23, 2 of 2 files at r24, 2 of 2 files at r25, 1 of 1 files at r26.
Review status: 5 of 9 files reviewed at latest revision, 25 unresolved discussions.

---

*[op_robot_tests/tests_files/base_keywords.robot, line 97 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggj-r10-97:-L6ReeNVh8196BxKYvl7:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/base_keywords.robot#L97)):*
<details><summary><i>Previously, kosaniak wrote…</i></summary><blockquote>

Run As
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/initial_data.py, line 153 at r23](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RHffq3pd4YRtgt_RY:-L6RHffrTYoRQ5Yxmd05:b-r26o22) ([raw file](https://github.com/openprocurement/robot_tests/blob/9a644e82636e30c26f1c4a8609229b302d2b59ee/op_robot_tests/tests_files/initial_data.py#L153)):*
> ```Python
> 
>     data['rectificationPeriod'] = {
>             "endDate": (get_now() + timedelta(minutes=(random.randint(5, 19)*1440)/accelerator)).isoformat()
> ```

Заміни `minutes=` на `days=` і забери `*1440`.

---

*[op_robot_tests/tests_files/keywords.robot, line 759 at r19](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXgYZHTRBkFCPhW-r19-759:-L6RSAuHVZH1-7ozZXhG:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/095c8354aa66ff2ac83c5abfc9404ae5f88cf256/op_robot_tests/tests_files/keywords.robot#L759)):*
<details><summary><i>Previously, ivanka12 wrote…</i></summary><blockquote>

для чого цей сліп, якщо функція wait_and_write_to_console вже виконує очікування і нічого не повертає?
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/keywords.robot, line 758 at r23](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RM3E9lp_soIwsoNYu:-L6RM3EAY9zcPArlIWMZ:b84qhbk) ([raw file](https://github.com/openprocurement/robot_tests/blob/9a644e82636e30c26f1c4a8609229b302d2b59ee/op_robot_tests/tests_files/keywords.robot#L758)):*
> ```RobotFramework
> Дочекатись дати
>   [Arguments]  ${date}
>   ${sleep}=  wait_and_write_to_console  ${date}
> ```

Можна обійтися без цього KW.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 932 at r19](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5JTrXfDDUESqvoeggk-r19-932:-L6NvtyG3-ESNBsAI3dZ:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/095c8354aa66ff2ac83c5abfc9404ae5f88cf256/op_robot_tests/tests_files/openProcedure.robot#L932)):*
<details><summary><i>Previously, ivanka12 wrote…</i></summary><blockquote>

provider1 заміни на provider2
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 329 at r23](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RSmRpJl4GAI4PUMkU:-L6RSmRqK9lK_RowJEcr:b-e3waen) ([raw file](https://github.com/openprocurement/robot_tests/blob/9a644e82636e30c26f1c4a8609229b302d2b59ee/op_robot_tests/tests_files/openProcedure.robot#L329)):*
> тендер
лот

---

*[op_robot_tests/tests_files/openProcedure.robot, line 333 at r23](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RT1ETOyE7-yP-7Rwq:-L6RT1EU_p9_szVWysU6:b-9yak9z) ([raw file](https://github.com/openprocurement/robot_tests/blob/9a644e82636e30c26f1c4a8609229b302d2b59ee/op_robot_tests/tests_files/openProcedure.robot#L333)):*
> ```RobotFramework
>   ...      ${USERS.users['${viewer}'].broker}
>   ...      tender_view
>   Отримати дані із поля rectificationPeriod.endDate тендера для усіх користувачів
> ```

Додай в service_keywords.py наступну функцію:

```
def compare_dates_by_day(left, right):
    return left.date() == right.date()
```

Потім додай сюди її виклик з перевіркою `Should Be True`, де left – `rectificationPeriod.endDate` із initial_data, а а right – щойно отримана дата.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 489 at r24](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RbSyH9nZa6Q0PosVJ:-L6RbSyIC0KLm47lkGPd:b-763rmt) ([raw file](https://github.com/openprocurement/robot_tests/blob/c06705e413d1371ceccfde391d90468cd3e2ae97/op_robot_tests/tests_files/openProcedure.robot#L489)):*
> level
тут і в решті місць зайве.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 734 at r24](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6Rc-tR5Ubml3Phwa8v:-L6Rc-tSzh_I4gXMlSox:b-92sdoo) ([raw file](https://github.com/openprocurement/robot_tests/blob/c06705e413d1371ceccfde391d90468cd3e2ae97/op_robot_tests/tests_files/openProcedure.robot#L734)):*
> ```RobotFramework
> Відображення опису нової номенклатури
>   [Tags]   ${USERS.users['${viewer}'].broker}: Відображення номенклатури лота
>   ...      viewer  tender_owner  provider  provider1
> ```

Додай provider2 і broker для нього.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 189 at r26](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RkPDeU-wpPEu9FJOs:-L6RkPDeU-wpPEu9FJOt:b-wuj3nn) ([raw file](https://github.com/openprocurement/robot_tests/blob/e3c00829c668beb31ccb8df71a76dea69a4765e2/op_robot_tests/tests_files/openProcedure.robot#L189)):*
> ```RobotFramework
> 
> 
> Відображення назви нас. пункту доставки активів лоту
> ```

`країни`

---

*[op_robot_tests/tests_files/openProcedure.robot, line 213 at r26](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RkWIr4MEQU9iUbXAD:-L6RkWIr4MEQU9iUbXAE:b-zfmbcq) ([raw file](https://github.com/openprocurement/robot_tests/blob/e3c00829c668beb31ccb8df71a76dea69a4765e2/op_robot_tests/tests_files/openProcedure.robot#L213)):*
> ```RobotFramework
> 
> 
> Відображення locality адреси доставки активів лоту
> ```

`населеного пункту`

---

*[op_robot_tests/tests_files/openProcedure.robot, line 836 at r26](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RhzlNFliavDGr2gOG:-L6RhzlNFliavDGr2gOH:b-92sdoo) ([raw file](https://github.com/openprocurement/robot_tests/blob/e3c00829c668beb31ccb8df71a76dea69a4765e2/op_robot_tests/tests_files/openProcedure.robot#L836)):*
> ```RobotFramework
> Відображення дати внесення останніх змін
>   [Tags]   ${USERS.users['${viewer}'].broker}: Відображення основних даних лоту
>   ...      viewer  tender_owner  provider  provider1
> ```

Додай provider2 і broker для нього.

---

*[op_robot_tests/tests_files/service_keywords.py, line 70 at r10](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L5OfEIAM0skiF30-5tl:-L6RRyIL78XVEQye_TZy:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/6ba5fc375faa96672fa42972a8f778a69b827393/op_robot_tests/tests_files/service_keywords.py#L70)):*
<details><summary><i>Previously, mykhaly (Yurii Mykhalchuk) wrote…</i></summary><blockquote>

>     if delta < seconds:
>         return False
>     return True
можна замінити на `return delta >= seconds`
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/service_keywords.py, line 62 at r23](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RRsE93-LDk088q9bz:-L6RRsE93-LDk088q9c-:b-a6c3dr) ([raw file](https://github.com/openprocurement/robot_tests/blob/9a644e82636e30c26f1c4a8609229b302d2b59ee/op_robot_tests/tests_files/service_keywords.py#L62)):*
> compare_dates

Поміняй їй назву.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests/742)*

**Comment by OrysiaDrabych** [comment]
Review status: 3 of 9 files reviewed at latest revision, 27 unresolved discussions.

---

*[op_robot_tests/tests_files/initial_data.py, line 153 at r23](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RHffq3pd4YRtgt_RY:-L6W0shsVokd4JfQWAOj:b-jm109l) ([raw file](https://github.com/openprocurement/robot_tests/blob/9a644e82636e30c26f1c4a8609229b302d2b59ee/op_robot_tests/tests_files/initial_data.py#L153)):*
<details><summary><i>Previously, selurvedu wrote…</i></summary><blockquote>

Заміни `minutes=` на `days=` і забери `*1440`.
</blockquote></details>

не вийде так. python2

---

*[op_robot_tests/tests_files/keywords.robot, line 758 at r23](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RM3E9lp_soIwsoNYu:-L6W16bVmLFwKDl6DRHM:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/9a644e82636e30c26f1c4a8609229b302d2b59ee/op_robot_tests/tests_files/keywords.robot#L758)):*
<details><summary><i>Previously, selurvedu wrote…</i></summary><blockquote>

Можна обійтися без цього KW.
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 329 at r23](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RSmRpJl4GAI4PUMkU:-L6W1DjfGhYDyWoRgMSi:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/9a644e82636e30c26f1c4a8609229b302d2b59ee/op_robot_tests/tests_files/openProcedure.robot#L329)):*
<details><summary><i>Previously, selurvedu wrote…</i></summary><blockquote>

> тендер
лот
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 333 at r23](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RT1ETOyE7-yP-7Rwq:-L6W1EAIc-t_JUW6xHPo:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/9a644e82636e30c26f1c4a8609229b302d2b59ee/op_robot_tests/tests_files/openProcedure.robot#L333)):*
<details><summary><i>Previously, selurvedu wrote…</i></summary><blockquote>

Додай в service_keywords.py наступну функцію:

```
def compare_dates_by_day(left, right):
    return left.date() == right.date()
```

Потім додай сюди її виклик з перевіркою `Should Be True`, де left – `rectificationPeriod.endDate` із initial_data, а а right – щойно отримана дата.

</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 489 at r24](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RbSyH9nZa6Q0PosVJ:-L6W1EV8JiAuGW3EPNCm:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/c06705e413d1371ceccfde391d90468cd3e2ae97/op_robot_tests/tests_files/openProcedure.robot#L489)):*
<details><summary><i>Previously, selurvedu wrote…</i></summary><blockquote>

> level
тут і в решті місць зайве.
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 734 at r24](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6Rc-tR5Ubml3Phwa8v:-L6W1FTpoqtA413Ax7mP:b-1gzmqz) ([raw file](https://github.com/openprocurement/robot_tests/blob/c06705e413d1371ceccfde391d90468cd3e2ae97/op_robot_tests/tests_files/openProcedure.robot#L734)):*
<details><summary><i>Previously, selurvedu wrote…</i></summary><blockquote>

Додай provider2 і broker для нього.
</blockquote></details>

В сценаріях тільки 2 учасники

---

*[op_robot_tests/tests_files/openProcedure.robot, line 189 at r26](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RkPDeU-wpPEu9FJOs:-L6W1N6Q1A6yGg4zCEnk:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/e3c00829c668beb31ccb8df71a76dea69a4765e2/op_robot_tests/tests_files/openProcedure.robot#L189)):*
<details><summary><i>Previously, selurvedu wrote…</i></summary><blockquote>

`країни`
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/openProcedure.robot, line 213 at r26](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RkWIr4MEQU9iUbXAD:-L6W1NQK5I9y0IHqxmEa:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/e3c00829c668beb31ccb8df71a76dea69a4765e2/op_robot_tests/tests_files/openProcedure.robot#L213)):*
<details><summary><i>Previously, selurvedu wrote…</i></summary><blockquote>

`населеного пункту`
</blockquote></details>

Done.

---

*[op_robot_tests/tests_files/service_keywords.py, line 62 at r23](https://reviewable.io:443/reviews/openprocurement/robot_tests/742#-L6RRsE93-LDk088q9bz:-L6W1R4mqnvZeGZ0tRO9:b-896fix) ([raw file](https://github.com/openprocurement/robot_tests/blob/9a644e82636e30c26f1c4a8609229b302d2b59ee/op_robot_tests/tests_files/service_keywords.py#L62)):*
<details><summary><i>Previously, selurvedu wrote…</i></summary><blockquote>

> compare_dates

Поміняй їй назву.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/robot_tests/742)*

**Pull request merged**